.github/workflows/cmake.yaml

```diff
@@ -164,6 +164,7 @@ jobs:
         run: |
           echo " -- switching to the pyre home directory"
           cd pyre
+          git pull --all
```

Comment:
you might be able to do this more efficiently with something like `git fetch ${GITHUB_REF}`

but I would probably suggest https://github.com/actions/checkout, it makes things much easier

Reply:
Thanks @rtburns-jpl, apparently we run into this known issue: https://github.com/actions/checkout/issues/23. This branch is for troubleshooting purposes and I had to open the PR only to run some tests.

Reply:
Speaking of which, the build matrix is a bit extreme. Any ideas on how to trim it to something that would catch most of the same errors at a fraction of the cost?

Reply:
> It may make some sense to run a reduced matrix on pull requests so the turnaround is quicker. This is probably best done by factoring the pull request trigger into its own action. Maybe pr-cmake so there is room for pr-mm. What do you think?

I was hoping the same fix would do for both the cmake and mm actions, as this issue has to do with the git checkout phase. As of now, I am only modifying `cmake.yaml`. Once we have figured it out, I will also modify `mm.yaml` in the same PR.

Reply:
> Speaking of which, the build matrix is a bit extreme. Any ideas on how to trim it to something that would catch most of the same errors at a fraction of the cost?

Perhaps, as a first attempt, the following matrix will do: cmake/mm, latest ubuntu, Debug/Release, latest gcc/clang.